sm/sm.c

```diff
@@ -59,6 +59,24 @@ void sm_sign(void* signature, const void* data, size_t len)
   sign(signature, data, len, sm_public_key, sm_private_key);
 }
 
+int sm_derive_sealing_key(unsigned char *key, const unsigned char *key_ident,
+                          size_t key_ident_size,
+                          const unsigned char *enclave_hash)
+{
+  unsigned char info[MDSIZE + key_ident_size];
+
+  memcpy(info, enclave_hash, MDSIZE);
+  memcpy(info + MDSIZE, key_ident, key_ident_size);
+
+  /*
+   * The key is derived without a salt because we have no entropy source
```

Comment:
You can use `platform_random()`. It's currently not hooked up to any entropy source but that's the API call for it.

Reply:
I think my comment is not entirely correct here because it is not the only problem with the salt. I can use `platform_random()` to generate a salt but this would break the functionality of the sealing feature. If the key is derived every time with a new random salt, the key differs from derivation to derivation. Then it is not possible for an enclave to derive the same key twice. 

I need to create the salt at the first sealing key derivation and then I have to store this salt somewhere on the platform in ways that it survives a restart of the security monitor. I have not found a solution to this problem, that is why I decided to derive the key without the salt.